### PR TITLE
Serde update

### PR DESCRIPTION
**Pull request by dmitrii-doronin**
### Pull request checklist

- [x] Did you bump the project version?
- [x] Did you add a description to the Pull Request?
- [x] Did you add new tests?
- [x] Did you add reviewers?

### CODE REVIEW

File: docs/src/index.md [hunk 1]
```diff
@@ -52,4 +52,12 @@ end
...
 # Serialize the JuliaCon instance to XML and print it
 to_xml(juliacon) |> print
+
+# Define serialization for JuliaCon struct to YAML format
+function Serde.SerYaml.ser_type(::Type{JuliaCon}, v::Date)
+    return Dates.format(v, "yyyy-mm-dd")
+end
+
+#  Serialize the JuliaCon instance to YAML and print it
+jo_yaml(juliacon) |> print
```
Comment (maxfadson) [inline docs/src/index.md]:
```suggestion
to_yaml(juliacon) |> print
```

File: test/Par/ParQuery.jl [hunk 2]
```diff
@@ -1,96 +1,216 @@
 # Par/ParQuery
 
-using Random: randstring
+using Serde
```
Comment (gryumov) [inline test/Par/ParQuery.jl]:
```suggestion
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
+## Serde.jl changelog
+
+The latest version of this file can be found at the master branch of the [Serde.jl](https://bhftbootcamp.github.io/Serde.jl) repository.
+
+### 0.2.2 (2024-22-02) ([merge request]())
```
Comment (gryumov) [inline CHANGELOG.md]:
```suggestion
### 2.0.0 (2024-22-02) ([merge request]())
```

File: src/Ser/SerJson.jl [hunk 4]
```diff
@@ -10,11 +10,10 @@ const INDENT = "  "
 
 # escaped string
 
-const NEEDESCAPE =
-    Set{UInt8}(UInt8['"', '\\', '\b', '\f', '\n', '\r', '\t'])
+const NEEDESCAPE = Set{UInt8}(UInt8['"', '\\', '\b', '\f', '\n', '\r', '\t'])
 
 function escape_char(b)
-    b == UInt8('"')  && return UInt8('"')
+    b == UInt8('"') && return UInt8('"')
```
Comment (gryumov) [inline src/Ser/SerJson.jl]:
```suggestion
    b == UInt8('"')  && return UInt8('"')
```

File: src/Ser/SerYaml.jl [hunk 5]
```diff
@@ -10,11 +10,10 @@ const INDENT_TYPES = [Pair, AbstractDict, AbstractVector, Tuple, NamedTuple, Abs
 
 # escaped string
 
-const NEEDESCAPE =
-    Set{UInt8}(UInt8['"', '\\', '\b', '\f', '\n', '\r', '\t'])
+const NEEDESCAPE = Set{UInt8}(UInt8['"', '\\', '\b', '\f', '\n', '\r', '\t'])
 
 function escape_char(b)
-    b == UInt8('"')  && return UInt8('"')
+    b == UInt8('"') && return UInt8('"')
```
Comment (gryumov) [inline src/Ser/SerYaml.jl]:
```suggestion
    b == UInt8('"')  && return UInt8('"')
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+- function `Serde.deser_xml()`
+- function `Serde.parse_xml()`
+- function `Serde.to_yaml()`
+- function `Serde.parse_yaml()`
+- function `Serde.deser_yaml()`
+
+#### Changed (1 change)
+
+- Refactored tests
```
Comment (gryumov) [inline CHANGELOG.md]:
```suggestion
# Serde.jl Changelog

The latest version of this file can be found at the master branch of the [Serde.jl repository](https://bhftbootcamp.github.io/Serde.jl).

## 2.0.0 (2024-02-22) ([merge request]())

### Added (5 changes)

- Function `Serde.deser_xml` for deserializing XML.
- Function `Serde.parse_xml` for parsing XML.
- Function `Serde.to_yaml` for converting to YAML.
- Function `Serde.parse_yaml` for parsing YAML.
- Function `Serde.deser_yaml` for deserializing YAML.

### Changed (1 change)

- Refactored tests to improve maintainability and performance.
```

**Comment by dmitrii-doronin** [comment]
@maxfadson questions or suggestions?

**Comment by maxfadson** [comment]
> @maxfadson questions or suggestions?

I noticed that multi-line elements/arguments now have a comma after the last line - is this ok? It’s quite strange to see this after `kw...` for example, but it’s just my habits

**Comment by maxfadson** [comment]
@dmitrii-doronin 
Otherwise ok, I like it. Especially ParXml now looks much more readable and concise.
By the way, will there be a squash of commits into a single one?

**Comment by dmitrii-doronin** [comment]
> > @maxfadson questions or suggestions?
> 
> I noticed that multi-line elements/arguments now have a comma after the last line - is this ok? It’s quite strange to see this after Kave, for example, but it’s just my habits

It's quite common I guess. Golang does the same thing pretty much. But I don't think it's necessary in Julia.

**Comment by dmitrii-doronin** [comment]
> @dmitrii-doronin Otherwise ok, I like it. Especially ParXml now looks much more readable and concise. By the way, will there be a squash of commits into a single one?

I hope someone would clean up SerJson and SerXml one day. It's generally not that hard. Current implementations seem messy to me. 

Yes, let's squash.

**Comment by dmitrii-doronin** [comment]
@maxfadson How do you like the language in general?

**Comment by maxfadson** [comment]
> @maxfadson How do you like the language in general?

I like it, it was nice to understand the details and write on it. It would be interesting to see what the backend written on it looks like.

**Comment by dmitrii-doronin** [comment]
@maxfadson  lgtm!

**Comment by maxfadson** [comment]
@dmitrii-doronin  I've been working with popular Java frameworks lately, so the experience with Julia is like a breath of fresh air for me 😅

**Comment by dmitrii-doronin** [comment]
> @dmitrii-doronin I've been working with OOP on popular Java frameworks lately, so the experience with Julia is like a breath of fresh air for me 😅

@maxfadson I've worked a lot with C++ in scientific computing, so I understand your suffering) 

Btw, we can merge this now)

**Comment by maxfadson** [comment]
@dmitrii-doronin 
I’m still waiting for the squash to be done so that you can write your own comment on the commit, we discussed this above. Or should I merge and then bring everything to my three commits and one of yours?

**Comment by dmitrii-doronin** [comment]
@maxfadson does not matter that much. Let's do it that way

**Comment by maxfadson** [comment]
@dmitrii-doronin ok, sorry for the wait. I think we should have contacted directly somehow :)

**Pull request merged**